nrf24L01_code/bootloader/WriteApp.c

```diff
@@ -0,0 +1,383 @@
+					}
+					else if (rxData.command == COMM_GET_STATE)
+					{
+						memset(ackResponse, 0x00, BUFFER_LENGTH);
+						ackResponse[0] = BOOTLOADER_FLASH_DATA;
+						ackResponse[1] = '0' + currentPage;
+						__nrfRadio_LoadAckPayload(flashPipe, (uint8_t*)ackResponse, 2);
+						waitRx();
+						// We remain in this state
+						bootloader_state = BOOTLOADER_FLASH_DATA;
+					}
+					else
+						bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}
+				else
+				{
+					bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}
+				
+				break;
+			}
+			case BOOTLOADER_FLASH_PAGE_DONE:
+			{	
+				waitRx();				
+				
+				if (rxData.hasMessage)
+				{
+					if (rxData.command == COMM_NEXT_PAGE)
+					{
+						currentPage++;
+						memset(pageData, 0x00, PAGE_SIZE);
+						bootloader_state = BOOTLOADER_REC_DATA;
+					}
+					else if (rxData.command == COMM_CHECK_CKS)
+					{
+						recvCKS = (rxData.data[0] << 8) | (rxData.data[1]);
+						bootloader_state = BOOTLOADER_CHECK_CKS;
+					}
+					else if (rxData.command == COMM_GET_STATE)
+					{
+						memset(ackResponse, 0x00, BUFFER_LENGTH);
+						ackResponse[0] = BOOTLOADER_FLASH_PAGE_DONE;
+						ackResponse[1] = '0' + currentPage;
+						__nrfRadio_LoadAckPayload(flashPipe, (uint8_t*)ackResponse, 2);
+						waitRx();
+						// We remain in this state
+						bootloader_state = BOOTLOADER_FLASH_PAGE_DONE;
+					}
+					else
+					bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}
+				else
+				{
+					bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}				
+				break;
+			}
+			case BOOTLOADER_CHECK_CKS:
+			{
+				bootloader_state = BOOTLOADER_FLASH_END;
+				uint8_t dummy = 0;
+				cli();
+				eeprom_write_block ((void*)&dummy, (void*)DOWNLOAD_FLAG_ADDRESS, DOWNLOAD_FLAG_LENGTH);
+				sei();
+				/*
+				memset(ackResponse, 0x00, BUFFER_LENGTH);
+				ackResponse[0] = BOOTLOADER_CHECK_CKS;
+				ackResponse[1] = 0xAA;
+				__nrfRadio_LoadAckPayload(flashPipe, (uint8_t*)ackResponse, 2);
+				waitRx();
+						
+				if (rxData.hasMessage)
+				{
+					if (rxData.command == COMM_FINISH_FLASH)
+					{
+						uint8_t dummy = 0;
+						eeprom_write_block ((void*)&dummy, (void*)DOWNLOAD_FLAG_ADDRESS, DOWNLOAD_FLAG_LENGTH);
+					}					
+					else if (rxData.command == COMM_GET_STATE)
+					{
+						memset(ackResponse, 0x00, BUFFER_LENGTH);
+						ackResponse[0] = BOOTLOADER_CHECK_CKS;
+						ackResponse[1] = 0xAA;
+						__nrfRadio_LoadAckPayload(flashPipe, (uint8_t*)ackResponse, 2);
+						waitRx();
+						// We remain in this state
+						bootloader_state = BOOTLOADER_CHECK_CKS;
+					}
+					else
+						bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}
+				else
+				{
+					bootloader_state = BOOTLOADER_FLASH_ERROR;
+				}		
+				*/
+				break;
+			}
+			default:
+				bootloader_state = BOOTLOADER_FLASH_ERROR;
+		}		
+				
+	} while ((bootloader_state != BOOTLOADER_FLASH_END) && (bootloader_state != BOOTLOADER_FLASH_ERROR));
+		
+		
+	
+	if (bootloader_state == BOOTLOADER_FLASH_ERROR)
```

Comment:
we should add some error code based on what went wrong, so this will be send back to the programmer as a ACK response